src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/compute/v2/OpenStackComputePluginTest.java

```diff
@@ -1,858 +1,688 @@
-        String idKeyName = null;
-        ComputeOrder computeOrder = new ComputeOrder(null, null, null, "default", null,
-                bestCpu, bestMemory, bestDisk, imageId, null, publicKey, networksId);
-        JSONObject computeJson = generateJsonRequest(imageId, bestFlavorId, userData, idKeyName, responseNetworkIds,
-                idInstanceName);
-
-        mockGetFlavorsRequest(bestFlavorId, bestCpu, bestMemory, bestDisk);
-        Mockito.when(this.launchCommandGeneratorMock.createLaunchCommand(computeOrder)).thenReturn(userData);
-        Mockito.doReturn(idInstanceName).when(this.computePlugin).getRandomUUID();
-        Mockito.when(this.clientMock.doPostRequest(this.argString.capture(), this.argBodyString.capture(), this.argCloudToken.capture()
-        )).thenReturn(expectedInstanceIdJson);
+        Mockito.doReturn(getHardwareRequirementsList()).when(this.computePlugin)
+                .getHardwareRequirementsList();
+        List<String> flavorIds = getFlavorIdsFromHardwareRequirementsList();
+        String unCachedFlavorId = "uncached-flavor-id";
+        flavorIds.add(unCachedFlavorId);
+
+        Mockito.when(this.clientMock.doGetRequest(Mockito.anyString(), Mockito.eq(cloudUser)))
+                .thenReturn(createGetFlavorResponseJson(unCachedFlavorId));
+
+        PowerMockito.mockStatic(GetFlavorResponse.class);
+        BDDMockito.given(GetFlavorResponse.fromJson(Mockito.anyString())).willCallRealMethod();
 
         // exercise
-        String instanceId = this.computePlugin.requestInstance(computeOrder, this.openStackV3User);
+        this.computePlugin.detailFlavors(ANY_URL, cloudUser, flavorIds);
 
         // verify
-        Assert.assertEquals(computeEndpoint, this.argString.getValue());
-        Assert.assertEquals(this.openStackV3User, this.argCloudToken.getValue());
-        // ToDo: fix the assertion below
-        // JSONAssert.assertEquals(computeJson.toString(), this.argBodyString.getValue(), false);
-        Assert.assertEquals(expectedInstanceId, instanceId);
+        Mockito.verify(computePlugin, Mockito.times(testUtils.RUN_ONCE))
+                .getHardwareRequirementsList();
+
+        Mockito.verify(this.clientMock, Mockito.times(testUtils.RUN_ONCE))
+                .doGetRequest(Mockito.anyString(), Mockito.eq(cloudUser));
+
+        PowerMockito.verifyStatic(GetFlavorResponse.class);
+        GetFlavorResponse.fromJson(Mockito.anyString());
     }
 
-    // test case: RequestInstance should throw InvalidParameter when KeyName post returns Bad Request
-    @Test(expected = InvalidParameterException.class)
-    public void testRequestInstanceOnBadRequestKeyNamePost() throws IOException, FogbowException {
+    // test case: When performing an unsuccessful request, it must re-wrap the
+    // HttpResponseException into an UnexpectedException
+    @Test(expected = UnexpectedException.class)
+    public void testDetailFlavorsUnsuccessful() throws FogbowException, HttpResponseException {
```

Comment:
same thing. You gotta verify if the mapper has been called. I think this method is too large, couldn't it be improved?

Reply:
do the same thing to the other methods that calls the exception mapper